use packaged pytest, not some venv

evgeni opened the pull request:
### Problem Statement

The current FAM test setup relies on being able to reach the Internet, which is "bad design"™ but also actually broken in IPv6-only environments because our lab network has IPv6, but no route to the broader Internet.

### Solution

Instead of installing test dependencies via `pip` from the Internet, install them via `dnf` from the RH CDN and use them in the tests here.
There is an accompanying PR against the repo that carries the setup playbooks.

### Related Issues


<!-- ### PRT test Cases example
trigger: test-robottelo
pytest: tests/foreman/ui/test_contenthost.py -k 'test_syspurpose_mismatched'
-->
<!--
PRT usage reference link: https://github.com/SatelliteQE/robottelo/wiki/Robottelo-Pull-Request-Testing-(PRT)-Process#usage-examples
-->

jameerpathan111 commented:
trigger: test-robottelo
pytest: tests/foreman/sys/test_fam.py
network_type: ipv6

evgeni commented:
@jameerpathan111 given the executed `fam-test-setup` playbook is the same for every Satellite branch, I think this needs cherry-picking to all supported branches.

Satellite-QE commented:
**PRT Result**
```
Build Number: 8994
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/sys/test_fam.py --external-logging
Test Result : =========== 108 failed, 2 passed, 472 warnings in 2113.86s (0:35:13) ===========
```

evgeni commented:
The PRT failure is expected, you need to run it against an AAP that has my deployment patches in.

jameerpathan111 commented:
@evgeni Acked. Can you fix pre-commit failures?

evgeni commented:
Done!

ekohl commented on tests/foreman/sys/test_fam.py:
```diff
@@ -74,6 +74,13 @@ def setup_fam(module_target_sat, module_sca_manifest, install_import_ansible_rol
     # Edit Makefile to not try to rebuild the collection when tests run
     module_target_sat.execute(f"sed -i '/^live/ s/$(MANIFEST)//' {FAM_ROOT_DIR}/Makefile")
```
Not a blocker, but could we introduce an environment variable for this as well so we can avoid patching code altogether?

evgeni replied:
Maybe, probably. But not today ;)

evgeni commented:
Rebased and updated to match the last version of https://github.com/theforeman/foreman-ansible-modules/pull/1788 -- now only replacing `pytest` to `$(PYTEST_COMMAND)` and letting the rest of the `-vv` handling to other means.
